src/crimson/osd/shard_services.cc

```diff
@@ -421,19 +426,23 @@ seastar::future<> OSDSingletonState::store_maps(ceph::os::Transaction& t,
       if (auto p = m->maps.find(e); p != m->maps.end()) {
 	auto o = std::make_unique<OSDMap>();
 	o->decode(p->second);
-	logger().info("store_maps osdmap.{}", e);
+	logger().info("store_maps storing osdmap.{}", e);
 	store_map_bl(t, e, std::move(std::move(p->second)));
 	osdmaps.insert(e, std::move(o));
 	return seastar::now();
       } else if (auto p = m->incremental_maps.find(e);
 		 p != m->incremental_maps.end()) {
+	ceph_assert (std::cmp_greater(e, 0u));
+	logger().info("store_maps found osdmap.{} incremental map, "
```

Comment:
Does it make sense to log before assertion?

Reply:
Yes, updated.